### ISSUE TITLE
Possible version mismatch in CoreFoundation.framework

### ISSUE THREAD

**Issue by betamos** [h4llow3En/mac-notification-sys#45]
See https://github.com/tauri-apps/tauri/issues/4387

I built a tauri app which includes mac-notification-sys, and one that doesn't. Both were built from macOS 12.4 with XCode 13 and 14 beta (no difference). It appears that our usage of this lib caused a "missing symbol" dyld linking error when loading the app on macOS 10.13-10.13.6 (at minimum, might affect more versions), but I don't know if it's a bug in tauri or in the build step here. So consider this issue informational for now.

The reason I'm posting here at all is because Tauri has a lot of Obj C dependencies, but this is the only one that appears to trigger a linking failure like this. 

Anyway, if anyone has experience deploying to multiple versions of macOS, this could be extremely helpful. I suspect there's a version mismatch with dynamic linking to libraries, but this is over my head. Nevertheless I really hope we can fix this.

**Comment by BlackHoleFox** [comment]
To prefix, I have no idea what `NSConstantDictionary` is because Apple doesn't document it or list it as a public API, but if this [dev forum post](https://developer.apple.com/forums/thread/683159) is to be believed, its a type that's created with the `@{@"` objective-C dictionary literal format. This crate uses that [a lot](https://github.com/h4llow3En/mac-notification-sys/blob/master/objc/notify.h#L70). Furthering this theory, I have a very small test binary whose only objective-C dependencies is this crate. Looking at the symbol list for this binary...
```
 nm ./target/debug/leak_hunting | rg _OBJC_CLASS_
                 U _OBJC_CLASS_$_NSBundle
                 U _OBJC_CLASS_$_NSConstantDictionary
                 U _OBJC_CLASS_$_NSConstantIntegerNumber
                 U _OBJC_CLASS_$_NSDate
                 U _OBJC_CLASS_$_NSDictionary
                 U _OBJC_CLASS_$_NSImage
                 U _OBJC_CLASS_$_NSObject
                 U _OBJC_CLASS_$_NSRunLoop
                 U _OBJC_CLASS_$_NSString
                 U _OBJC_CLASS_$_NSThread
                 U _OBJC_CLASS_$_NSURL
                 U _OBJC_CLASS_$_NSUserNotification
                 U _OBJC_CLASS_$_NSUserNotificationCenter
                 U _OBJC_CLASS_$_NSWorkspace
000000010006c758 S _OBJC_CLASS_$_NotificationCenterDelegate
000000010006c398 s __OBJC_CLASS_PROTOCOLS_$_NotificationCenterDelegate
000000010006c4a8 s __OBJC_CLASS_RO_$_NotificationCenterDelegate
```

... its plain to see that `NSConstantDictionary` is registered in the symbol table, which means that it needs to be available for the app to run. If I go through and comment out all of those dictionary initializers and run it again, the symbol goes poof:
<img width="507" alt="image" src="https://user-images.githubusercontent.com/20936452/179382474-07a160fc-5a94-4486-a24e-dfb3ffa9741d.png">

So the answer is probably "yes, this crate is causing it", but more investigation should be done to determine why this symbol doesn't exist on macOS 10.14. XCode thinks that the syntax is available on a 10.14 deployment target so maybe the way this crate's objective-C is being built is not complete enough.

As a quick wrap up, I built a objective-C CLI with XCode and a deployment target of 10.14. Here's the source:
```objectivec
#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        // insert code here...
        NSLog(@"Hello, World!");
    }
        
    NSDictionary *value = @{
        @"helloString" : @"Hello, World!",
        @"magicNumber" : @42,
    };
    
    return 0;
}
```

and here are the symbols. XCode seems to be doing something different that results in the `NSConstantDictionary` object not being used. Unsure of anything more than that though.
```
nm ./Products/usr/local/bin/versiontester | rg _OBJC_CLASS_
                 U _OBJC_CLASS_$_NSDictionary
                 U _OBJC_CLASS_$_NSNumber
```

**Comment by betamos** [comment]
Wow, thanks a ton! I didn't expect this to be paid attention to, and especially not that fast. Your analysis looks spot on.

It appears the PR fixes the issue but I'll share the details there instead.

Thanks again.
